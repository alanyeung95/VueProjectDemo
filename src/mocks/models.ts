export function mockDemoMessage(): string {
  return "Demo message";
}
