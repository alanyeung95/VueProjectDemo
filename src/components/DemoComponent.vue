<template>
  <div class="hello">
    <h1>{{ testingMessage }}</h1>
    <h1>{{ msgFromParent }}</h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class HelloWorld extends Vue {
  @Prop() private msgFromParent!: string;

  get testingMessage() {
    return "computed message";
  }
}
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
