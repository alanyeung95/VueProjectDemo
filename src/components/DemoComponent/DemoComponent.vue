<template>
  <div class="hello">
    <h1>{{ testingMessage }}</h1>
    <h1>{{ msgFromParent }}</h1>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class DemoComponent extends Vue {
  @Prop() private msgFromParent!: string

  get testingMessage() {
    return 'Hi (computed value) '
  }
}
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
