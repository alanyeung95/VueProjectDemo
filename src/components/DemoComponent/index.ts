import DemoComponent from "./DemoComponent.vue";

export default DemoComponent;
