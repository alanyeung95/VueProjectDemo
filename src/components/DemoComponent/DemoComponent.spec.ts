import { Wrapper, shallowMount } from "@vue/test-utils";
import DemoComponent from "./DemoComponent.vue";
import { mockDemoMessage } from "@/mocks/models";

describe("DemoComponent.vue", () => {
  test("renders props.msg when passed", () => {
    const mockMessage = mockDemoMessage();
    const wrapper = shallowMount(DemoComponent, {
      propsData: { mockMessage },
    });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
