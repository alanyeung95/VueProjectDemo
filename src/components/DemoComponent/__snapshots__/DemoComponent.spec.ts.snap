// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoComponent.vue renders props.msg when passed 1`] = `
<div
  class="hello"
  mockmessage="Demo message"
>
  <h1>
    computed message
  </h1>
   
  <h1>
    
  </h1>
</div>
`;
